.pre-commit-config.yaml

```diff
@@ -20,6 +20,15 @@ repos:
     hooks:
       - id: black
         exclude: docs/.*|tests/.*.fixture|.*.ge_store_backend_id
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.1.0
+    hooks:
+      - id: codespell
+        args: 
+          - --ignore-words-list
+          - ba,cros,fo,hist,nd,nin,onl,te,tring,wont
+          - --skip
+          - *.csv,*.lock
```

Comment:
While I do very much like this tool, I do believe adding it to our pre-commit config is out of the scope of this current PR. 

Due to the high frequency of commits made by our developers and contributors from the GE community, I'd be wary of adding additional complexity to what is specifically designed to be a lightweight process. 

This is something that perhaps requires further discussion and thought from our developer community. If you feel strongly about this addition, could you please port it over to an issue so it may be evaluated separately from the changes made in this PR?